chore(VariantGenotypeQualityMetrics): make more dry

ignatiusm opened the pull request:
This PR contains a small refactor of `VariantGenotypeQualityMetrics.tsc` to make it more DRY (related to issue #1154).

I have just changed my dev environment (from Linux to MacOS), and am having issues running the tests, but a visual check comparing the Broad-deployed Variant page chart against a locally-served version of the page suggests that the functionality is the same for an ad-hoc selection of variants. I figured this would be good enough for you to give it an initial review while I sort out running the tests.

Of course, happy to hear any feedback - don't hold back with suggestions! Similarly, as per discussion in #1154,
if this seems like it will be more hassle for you than it is worth then feel free to ask me to stop :)

ignatiusm commented:
@rileyhgrant - FYI

rileyhgrant reviewed (changes requested):
Thanks so much for doing this @ignatiusm, this looks great. We really appreciate you taking the time to do this (and for your patience during the holiday season as this PR was mostly untouched by us!). Happy new year!

A few comments on the PR, all minor and small
- one comment and possible follow up nitpick re: a TS `Interface` vs `Type`
- Re: CI and tests, it seems as though we currently need to approve CI runs for new PRs from outside collaborators, now that it's been run, there are a few CI steps that need to be appeased. These can be found in the frontend CI file (`/.github/workflows/browser-ci.yml`), but this brings to light that we on the gnomAD browser team should document running the tests locally and possibly make it more convenient to do so
  - For now, you should be able to run (from the root of the project): 
    - `yarn run eslint .` and 
    - `yarn run jest`
      - Since a snapshot test will now fail (expectedly, since the tests are rendering a more properly mocked table after the refactor) the `-u` flag will be used to update failing snapshot(s).
 
Those should be the two steps most relevant to the changes you made, `jest` does funny things when run from the `browser` subdirectory.

I'm also happy to add commits that appease the CI, if you think that makes more sense, as those are pretty chore-like. Just figured I'd mention this here also.

Thanks again for doing this, the dry-ing up looks great and we very much appreciate your attention to detail on manually checking the functionally on locally served page vs production.

ignatiusm commented on browser/src/VariantPage/VariantGenotypeQualityMetrics.tsx:
```diff
@@ -71,6 +71,112 @@ type VariantGenotypeQualityMetricsProps = {
   variant: Variant
 }
 
+interface Tab {
```
Resolved in [813fa9d](https://github.com/broadinstitute/gnomad-browser/pull/1354/commits/813fa9d49da8d89c51134ca65f499cc94ce3c699)

ignatiusm commented:
Thanks @rileyhgrant, Happy New Year to you too :) :tada:

I've addressed your comments by:
- changing the `interface` declaration to `type`;
- updating the snapshots; and,
- making changes to get `eslint` passing.

Let me know if there are other changes you'd like to see.

ignatiusm commented:
Ah I see the CI has failed (and I forgot to go through `/.github/workflows/browser-ci.yml` like you suggested). I'll look at this today.

rileyhgrant commented:
Thanks for your continued work and your patience on this one @ignatiusm.

I'll plan on adding better documentation on running the test suite that the frontend CI goes through locally, so that the back and forth can be avoided a bit in the future.

ignatiusm commented:
"All checks have passed" :tada: Can you please review my recent changes @rileyhgrant? Getting closer :smile:

rileyhgrant reviewed (approved):
Woohoo, LGTM!

Thanks again so much for doing this, @ignatiusm :)

ignatiusm commented:
Thanks @rileyhgrant ! Is it okay for me to work through that priority list in #1154? I'll create a separate PR for each item. Plus I know how to run the tests now, so reviewing should be easier :laughing:

rileyhgrant commented:
> Thanks @rileyhgrant ! Is it okay for me to work through that priority list in #1154? I'll create a separate PR for logical block of refactors (e.g. there are a few related to the `ShortTandemRepeat` page and it's components). Plus I know how to run the tests now, so reviewing should be easier 😆

Yes, if you're willing, absolutely! :)
